task10/view/feedView.js

```diff
@@ -0,0 +1,159 @@
+        this._btn.style.visibility = 'hidden';
+    }
+
+    clearFeed() {
+        this._btn = this._btn.cloneNode(true);
+        this._btn.addEventListener('click', mainController.loadTweets);
+        this._postContainer.innerHTML = '';
+        this._postContainer.appendChild(this._btn);
+    }
+
+    addItem(post) {
+        let newPost = document.importNode(this._postTemplate.content, true);
+        this._fillItemData(newPost, post);
+        this._postContainer.insertBefore(newPost, this._btn);
+    }
+
+    _fillItemData(item, data) {
+        let placeholders = item.querySelectorAll('[data-target]');
+        [].forEach.call(placeholders || [], (phElement) => {
+            let key = phElement.getAttribute('data-target');
+
+            if (key === 'likes') {
+                phElement.textContent = String(data[key].length);
+                this._checkForCurUser(phElement, data[key]);
+            }
+            else if (key === 'hashTags') this._addTags(phElement, data[key]);
+            else phElement.textContent = String(data[key] || '');
+        });
+        item.querySelector('.post').setAttribute('id', data['id']);
+        let bot = item.querySelector('.bot');
+
+        if (this._user === data['author']) this._addButtons(bot);
+
+        this._insertPhoto(item, data.photoLink, bot);
+        //<button class="del">Delete</button>
+        //<button class="edit">Edit</button>
+    }
+
+    _checkForCurUser(likeCountPh, arr) {
+        if(arr.includes(this._user)) this._toggleIcon(likeCountPh);
+    }
+
+    _toggleIcon(likes) {
+        let likeIcon = likes.parentElement.getElementsByTagName('i')[0];
+        this._toggleStyle(likeIcon);
+    }
+
+    _toggleStyle(likeIcon) {
+        likeIcon.classList.toggle('fas');
+        likeIcon.classList.toggle('far');
+        likeIcon.classList.toggle('active');
+    }
+
+    _addTags(item, tags) {
+        tags.forEach((tag) => { item.appendChild(this._makeTag(tag)); });
+    }
+
+    _makeTag(tag) {
+        let newTag = document.importNode(this._tagTemplate.content, true);
+        newTag.querySelector('.tag').textContent = tag;
+        return newTag;
+    }
+
+    _insertPhoto(item, link, bot) {
+        if((link || '') !== '')
+            item.querySelector('.post').insertBefore(this._createImg(link), bot);
+    }
+
+    _createImg(link) {
+        let img = document.createElement('img');
+        img.setAttribute('class', 'att');
+        img.setAttribute('src', link);
+        img.setAttribute('alt', 'Image');
+        return img;
+    }
+
+    _addButtons(bot) {
+        let button = document.createElement('button');
+        button.setAttribute('class', 'del');
+        button.textContent = 'Delete';
+        bot.appendChild(button);
+        button = document.createElement('button');
+        button.setAttribute('class', 'edit');
+        button.textContent = 'Edit';
+        bot.appendChild(button);
+    }
+
+    removeItem(id) {
+        if(confirm('Are you sure?'))
```

Comment:
It will be removed from controller, but not from View if user press cancel
![bug](https://user-images.githubusercontent.com/25108879/83197888-e9c4e880-a146-11ea-90b5-7ed9e427c090.gif)